### PR TITLE
Smart contract overview update

### PR DESCRIPTION
**Pull request by sbvegan**
**Description**

- Bringing the smart contract overview to the top level to improve visibility
- Brought the releases to the top of the page
- Added the changelog of the contracts
- Updated the links to the latest updated version for each contract
- Moved the deprecated contracts to the bottom of the section
- Added missing core contracts
- Added redirect

### CODE REVIEW

**Comment by sbvegan** [comment]
The broken links are unrelated to this PR and because there are so many I think it should be addressed in a different one

File: pages/stack/smart-contracts.mdx [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+
+The layer 1 contracts of the OP Stack are deployed on Ethereum. Their primary
+purpose is to facilitate the cross domain message passing and maintain the
+valid state root of the layer 2.
+
+### Official Releases
+
+The full smart contract release process is documented in the [monorepo](https://github.com/ethereum-optimism/optimism?tab=readme-ov-file#development-and-release-process).
+All production releases are always tags, versioned as `<component-name>/v<semver>`
+and contract releases you'll see them tagged as `op-contract/vX.X.X`.
```
Comment (OPMattie) [inline pages/stack/smart-contracts.mdx:23]: Non-blocking light edits, bolding for visibility but that formatting shouldn't be carried over

> All production releases are always tagged, versioned as `<component-name>/v<semver>`
> and contract releases you'll see tagged as `op-contract/vX.X.X`.

File: pages/stack/smart-contracts.mdx [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+  For contract releases, refer to the GitHub release notes for a given release,
+  which will list the specific contracts being released—**not all contracts are
+  considered production ready within a release**, and many are under active
+  development. These release pages are linked below.
+
+  Tags of the form `v<semver>`, such as `v1.1.4`, indicate releases of all Go code
+  only, and **DO NOT** include smart contracts.
+</Callout>
+
+#### op-contracts/v1.5.0 - Safe Extensions
```
Comment (OPMattie) [inline pages/stack/smart-contracts.mdx:35]: @sbvegan thoughts on adding a callout here for who should / should not use these, even though they're gov approved? The callout would point to op-contracts/v1.3.0 with something like, "Note: While these have been gov-approved, the recommended release for new production chains is..."

File: pages/stack/smart-contracts.mdx [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+  *   DelayedWETH: [1.0.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol#L25)
+  *   MIPS: [1.0.1](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/src/cannon/MIPS.sol#L47)
+  *   PreimageOracle: [1.0.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/src/cannon/PreimageOracle.sol#L33)
+  *   SuperchainConfig: [1.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/SuperchainConfig.sol#L38)
+  *   ProtocolVersions: [1.0.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/ProtocolVersions.sol#L39)
+
+</details>
+
+#### op-contracts/v1.4.0 - Fault Proofs
+
```
Comment (OPMattie) [inline pages/stack/smart-contracts.mdx:86]: Likewise consider a callout here

File: pages/stack/smart-contracts.mdx [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+  *   L1CrossDomainMessenger: [2.3.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol#L24)
+  *   L1ERC721Bridge: [2.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol#L27)
+  *   L1StandardBridge: [2.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L1StandardBridge.sol#L73)
+  *   OptimismMintableERC20Factory: [1.9.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/universal/OptimismMintableERC20Factory.sol#L46)
+  *   SuperchainConfig: [1.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/SuperchainConfig.sol#L38)
+  *   ProtocolVersions: [1.0.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/ProtocolVersions.sol#L39)
+</details>
+
+#### op-contracts/v1.3.0 - Multi-Chain Prep (MCP)
+
```
Comment (OPMattie) [inline pages/stack/smart-contracts.mdx:137]: Callout for this is the recommended contract set for new chains

**Review by OPMattie — APPROVED** [review]

File: pages/stack/smart-contracts.mdx [hunk 5]
```diff
@@ -0,0 +1,625 @@
...
+</details>
+
+### L1 Contract Details
+
+#### AddressManager [Latest (this has no version)](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/legacy/AddressManager.sol)
+
+AddressManager is a legacy contract that was used in the old version of the 
+Optimism system to manage a registry of string names to addresses. We now use a
+more standard proxy system instead, but this contract is still necessary for 
+backwards compatibility with several older contracts.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:270]: All existing and new deployments will still use this—the L1CrossDomainMessenger is always behind a ResolvedDelegateProxy, and ResolvedDelegateProxy is the proxy type we have that relies on an AddressManager. Feel free to tweak phrasing how you see fit
```suggestion
AddressManager is a legacy contract that was used in the old version of the 
Optimism system to manage a registry of string names to addresses. We now use a
more standard proxy system for most contracts, but this contract is still used for
the L1CrossDomainMessenger, via ResolvedDelegateProxy
```

File: pages/stack/smart-contracts.mdx [hunk 6]
```diff
@@ -0,0 +1,625 @@
...
+
+#### SuperchainConfig: [1.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/SuperchainConfig.sol#L38)
+
+The SuperchainConfig contract is used to manage configuration of global 
+superchain values.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:275]: The key thing it currently handles is pausing/unpausing all withdrawals (ETH/ERC20/ERC721) for the superchain, which I think is worth calling out here.

File: pages/stack/smart-contracts.mdx [hunk 7]
```diff
@@ -0,0 +1,625 @@
...
+#### L1CrossDomainMessenger: [2.3.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol#L24)
+
+The L1CrossDomainMessenger is a message passing interface between L1 and L2
+responsible for sending and receiving data on the L1 side. Users are encouraged
+to use this interface instead of interacting with lower-level contracts directly.
+
+#### L1ERC721Bridge: [2.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol#L27)
+
+The `L1ERC721bridge` is a contract which works together with the L2 ERC721 bridge
+to make it possible to transfer ERC721 tokens from Ethereum to Optimism. This
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:286]: I think our docs are OP Mainnet specific, ignore if not
```suggestion
to make it possible to transfer ERC721 tokens from Ethereum to OP Mainnet. This
```

File: pages/stack/smart-contracts.mdx [hunk 8]
```diff
@@ -0,0 +1,625 @@
...
+  This contract is not intended to support all variations of ERC20 tokens. Examples
+  of some token types that may not be properly supported by this contract include, but are
+  not limited to: tokens with transfer fees, rebasing tokens, and tokens with blocklists.
+</Callout>
+
+The `L1StandardBridge` is responsible for transferring ETH and ERC20 tokens between L1 and
+L2. In the case that an ERC20 token is native to L1, it will be escrowed within this
+contract. If the ERC20 token is native to L2, it will be burnt. Before Bedrock, ETH was
+stored within this contract. After Bedrock, ETH is instead stored inside the
+OptimismPortal contract.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:301]: There is not ETH in this contract on L1, so maybe we delete the pre-Bedrock note for simplicity?

File: pages/stack/smart-contracts.mdx [hunk 9]
```diff
@@ -0,0 +1,625 @@
...
+
+#### OptimismPortal: [3.10.0](https://github.com/ethereum-optimism/optimism/blob/547ea72d9849e13ce169fd31df0f9197651b3f86/packages/contracts-bedrock/src/L1/OptimismPortal2.sol#L144)
+
+The `OptimismPortal` is a low-level contract responsible for passing messages between L1
+and L2. Messages sent directly to the `OptimismPortal` have no form of replayability.
+Users are encouraged to use the `L1CrossDomainMessenger` for a higher-level interface.
+
+#### ProtocolVersions: [1.0.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/ProtocolVersions.sol#L39)
+
+The `ProtocolVersions` contract is used to manage Superchain protocol version information.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:311]: If you want to expand, there's more info here: https://specs.optimism.io/protocol/superchain-upgrades.html

The gist is that it exposes a `recommended` and `required` node version (the go code releases, NOT contract releases) for node operators. The recommended/required versions are changed with each hard fork. Nodes can optionally halt if using the wrong version

File: pages/stack/smart-contracts.mdx [hunk 10]
```diff
@@ -0,0 +1,625 @@
...
+
+#### SuperchainConfig: [1.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0/packages/contracts-bedrock/src/L1/SuperchainConfig.sol#L38)
+
+The `SuperchainConfig` contract is used to manage configuration of global Superchain values.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:315]: This was documented above also, it's duplicated here

File: pages/stack/smart-contracts.mdx [hunk 11]
```diff
@@ -0,0 +1,625 @@
...
+
+#### SystemConfig: [2.0.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/src/L1/SystemConfig.sol#L111)
+
+The `SystemConfig` contract is used to manage configuration of an OP Stack network.
+All configuration is stored on L1 and picked up by L2 as part of the derivation of
+the L2 chain.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:321]: It's not *all* configuration, so we may want to tweak the phrasing a bit. Also a nice aspect of the SystemConfig (starting from MCP) is that it has references to all other contract addresses for the chain, so just by having the system config address for a chain you can discover the rest of its contracts

File: pages/stack/smart-contracts.mdx [hunk 12]
```diff
@@ -0,0 +1,625 @@
...
+
+#### DEPRECATED - L2OutputOracle: [1.8.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/src/L1/L2OutputOracle.sol#L59)
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:323]: As mentioned above, we also need to add docs for the new FP contracts in this section

File: pages/builders/chain-operators/self-hosted.mdx [hunk 13]
```diff
@@ -20,7 +20,7 @@ There are two main steps to get started building your own self-hosted OP Chain:
   To work with OP Chains, you'll need to understand the fundamental components of OP Chains.
 
   *   **Chain Architecture**: OP Chains use execution and consensus clients as well as the OP Stack's privileged roles. For more details, see the [Chain Architecture](/builders/chain-operators/architecture) guide.
-  *   **Smart Contracts**: OP Chains use several smart contracts on the L1 blockchain to manage aspects of the Rollup. Each OP Stack chain has its own set of [L1 smart contracts](/stack/protocol/rollup/smart-contracts#l1-contracts) and [L2 contracts or predeploys](/stack/protocol/rollup/smart-contracts#l2-contracts-predeploys) that are deployed when the chain is created.
+  *   **Smart Contracts**: OP Chains use several smart contracts on the L1 blockchain to manage aspects of the Rollup. Each OP Stack chain has its own set of [L1 smart contracts](/stack/smart-contracts#layer-1-contracts) and [L2 predeploy contracts](/stack/smart-contracts#layer-2-contracts-predeploys) that are deployed when the chain is created.
```
[the remark this answers is not in the document]
Comment (sbvegan) [reply pages/builders/chain-operators/self-hosted.mdx:23]: good call

File: pages/stack/smart-contracts.mdx [hunk 14]
```diff
@@ -0,0 +1,625 @@
...
+  *   L1ERC721Bridge: [1.1.1](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.0.0/packages/contracts-bedrock/contracts/L1/L1ERC721Bridge.sol#L23)
+  *   L1StandardBridge: [1.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.0.0/packages/contracts-bedrock/contracts/L1/L1StandardBridge.sol#L94)
+  *   L2OutputOracle: [1.3.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.0.0/packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol#L81)
+  *   OptimismMintableERC20Factory: [1.1.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.0.0/packages/contracts-bedrock/contracts/universal/OptimismMintableERC20Factory.sol#L47)
+  *   OptimismPortal: [1.6.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.0.0/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol#L143)
+  *   SystemConfig: [1.3.0](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.0.0/packages/contracts-bedrock/contracts/L1/SystemConfig.sol#L83)
+</details>
+
+### L1 Contract Details
```
[the remark this answers is not in the document]
Comment (sbvegan) [reply pages/stack/smart-contracts.mdx:264]: Good idea, I'll remove them and folks can use the links in the changelog. I'll also track down those missing contracts

File: pages/stack/smart-contracts.mdx [hunk 15]
```diff
@@ -0,0 +1,679 @@
...
+
+#### DisputeGameFactory
+
+The `DisputeGameFactory` contract generates `FaultDisputeGame` contract 
+instances.
+
+#### FaultDisputeGame 
+
+The `FaultDisputeGame` contract instances that each act as a host to a proposal 
+about the state of the OP Stack chain at a given block number.
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:345]: Typo in this sentence? "The `FaultDisputeGame` contract instances that each act" is confusing me

File: pages/stack/smart-contracts.mdx [hunk 16]
```diff
@@ -0,0 +1,679 @@
...
+
+### GasPriceOracle
+
+In the legacy system, the `GasPriceOracle` was a permissioned contract that was
+pushed the L1 base fee and the L2 gas price by an offchain actor. The offchain
+actor observes the L1 blockheaders to get the L1 base fee as well as the gas
+usage on L2 to compute what the L2 gas price should be based on a congestion
+control algorithm.
+
+After Bedrock, the `GasPriceOracle` is no longer a permissioned contract and
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:465]: Delete legacy pre-Bedrock info and just explain what it currently is?

File: pages/stack/smart-contracts.mdx [hunk 17]
```diff
@@ -0,0 +1,679 @@
...
+*   **Address:** `0x4200000000000000000000000000000000000015`
+*   **Introduced:** Bedrock
+*   **Deprecated:** no
+*   **Proxied:** yes
+
+### L2ToL1MessagePasser
+
+<Callout>
+  There is a legacy contract under this same name.
+</Callout>
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:528]: What does this mean? I'm not familiar with this (doesn't mean it's not true, just unclear what this means exactly and if this matters to readers)

File: pages/stack/smart-contracts.mdx [hunk 18]
```diff
@@ -0,0 +1,679 @@
...
+### L2ERC721Bridge
+
+<Callout type="warning">
+  Do not bridge an ERC721 that was originally deployed on Optimism. This
+  bridge ONLY supports ERC721s originally deployed on Ethereum. Users will need to
+  wait for the one-week challenge period to elapse before their Optimism-native NFT
+  can be refunded on L2.
+</Callout>
+
+The `L2ERC721Bridge` is a contract which works together with the L1 ERC721 bridge to
```
Comment (mds1) [inline pages/stack/smart-contracts.mdx:549]:
```suggestion
The `L2ERC721Bridge` is a contract which works together with the `L1ERC721Bridge` to
```

File: pages/stack/smart-contracts.mdx (hunk 15 above)
Comment (sbvegan) [reply pages/stack/smart-contracts.mdx:345]: Yeah this is a frankenstein's monster from pulling this out of the dipsutegamefactory

File: pages/stack/smart-contracts.mdx (hunk 17 above)
Comment (sbvegan) [reply pages/stack/smart-contracts.mdx:528]: idk where this came from tbh, I'm going to remove it

**Pull request merged**